engine/src/main.cpp

```diff
@@ -306,7 +306,10 @@ int main( int argc, char *argv[] )
     std::string program_path_str(argv[0]);
     boost::filesystem::path program_path(program_path_str);
```

Comment:
What if we were to add in here something like:

```C++
boost::filesystem::path abs_program_path = boost::filesystem::absolute(program_path);
```

And then the next line (308) would become:

```C++
boost::filesystem::path program_directory_path = abs_program_path.parent_path();
```

Then we could be reasonably sure that we had a non-empty path to cd to.

Reply:
We also probably only want to change directory there if some of the required data files exist at that path too.
So it'd be good to have a function that did the following:
1. Take a target path (absolute path, f.e `boost::filesystem::absolute()`)
2. Verify that the path contains the required data files at that location
3. Warn if no data files are present at that location, and return an error to the caller
4. Caller can then decide to try another path or error out entirely

Reply:
I've merged my small change. I really don't know enough about this boost stuff to properly implement Benjamin's suggestions. As far as I know, Vega Strike used to look for Version.txt, and if it did not exist, would error out.

Obviously the best way forward would be for one to be able to just type "vegastrike" anywhere in the file system, and the game to read your config file, figure out where the data is supposed to be and changing to that directory. 
There are still some messages about a game file directory supposed to be in the config file, but I have not seen the variable and do not know what it is supposed to look like. 

One other thing to think about is that vegastrike is a game engine that can run more than one game, that could conceivably be installed concurrently.

Reply:
@evertvorster agree - it really should be in the config file. There's a number of ways that could be tackled:

- by default it would look at one config file but for alternate games a different config file could should be specified, or
- get the executable base name (e.g vegastrike) and look for the config at a given path (`~/.local/<executable base name>/...); alternate games would simply symlink the binary to rename it to change the path.

Basically we do something like:
```c++
boost::filesystem::path program_path(program_path_str);
boost::filesystem::path version_file = progrgram_path + "/" + "Version.txt"
if (!version_file.exists()) // or `false == version_file.exists()`
{
   // generate error
}
// allow change directory to happen
```
See https://www.boost.org/doc/libs/1_74_0/libs/filesystem/doc/reference.html#exists